Frends.SFTP.UploadFiles/Frends.SFTP.UploadFiles/Definitions/SingleFileTransfer.cs

```diff
@@ -97,8 +99,9 @@ internal SingleFileTransferResult TransferSingleFile()
         return _result;
     }
 
-    private void GetSourceFile()
+    private void GetSourceFile(CancellationToken cancellationToken)
     {
+        cancellationToken.ThrowIfCancellationRequested();

```

Comment:
No need to check cancellation here.

Reply:
The Task has been known for getting stuck and cannot be terminated. This is the reason I added these cancellation checks in different layers of the Task.